packages/client/machines/appMachine.ts

```diff
@@ -26,102 +40,209 @@ export const createAppMachine = ({
                         },
+                    },
+                },
+
+                waitingForUserAuthentication: {
+                    tags: 'userIsUnauthenticated',
+
+                    type: 'parallel',
+
+                    states: {
+                        signingIn: {
+                            initial: 'waitingForCredentials',
+
+                            states: {
+                                waitingForCredentials: {},
+
+                                submitingCredentials: {
+                                    invoke: {
+                                        src: 'signIn',
 
-                        invoke: {
-                            id: 'fetchAcknowledgementStatus',
+                                        onDone: {
+                                            target: 'successfullySubmittedCredentials',
+                                        },
 
-                            src: () => (sendBack) => {
-                                socket.emit(
-                                    'GET_HAS_ACKNOWLEDGED_CONNECTION',
-                                    () => {
-                                        sendBack(
-                                            appModel.events.ACKNOWLEDGE_SOCKET_CONNECTION(),
-                                        );
+                                        onError: {
+                                            target: 'erredSubmittingCredentials',
+                                        },
                                     },
-                                );
+                                },
+
+                                successfullySubmittedCredentials: {
+                                    type: 'final',
+                                },
+
+                                erredSubmittingCredentials: {},
+                            },
+
+                            on: {
+                                SIGN_IN: {
+                                    target: 'signingIn.submitingCredentials',
+
+                                    actions: appModel.assign({
+                                        email: (_, event) => event.email,
+                                        password: (_, event) => event.password,
+                                    }),
+                                },
+                            },
+
+                            onDone: {
+                                target: '#app.waitingForServerToAcknowledgeSocketConnection',
+
+                                actions: 'reconnectSocket',
                             },
                         },
                     },
+                },
+
+                waitingForServerToAcknowledgeSocketConnection: {
+                    tags: ['showApplicationLoader', 'userIsAuthenticated'],
+
+                    initial: 'fetching',
+
+                    states: {
+                        fetching: {
+                            after: {
```

Comment:
❔ question: Why are you using a timer here too before bouncing ?

Reply:
This is not a recent code, but the timer in `fetching` state is here to cancel the service after 500ms, to prevent the machine to be stuck while waiting for a response from the server.